.github/workflows/build.yml

```diff
@@ -0,0 +1,118 @@
+name: Build
+
+on: [push, pull_request]
```

Comment:
The suggested default main.yml suggests a slightly more specific condition:
```yml
# Controls when the action will run. 
on:
  # Triggers the workflow on push or pull request events but only for the mega branch
  push:
    branches: [ mega ]
  pull_request:
    branches: [ mega ]
```

Reply:
btw I never pushed this change, workflow applies to any branch created on the repo